src/main/java/homework16/Product.java

```diff
@@ -0,0 +1,136 @@
+package homework16;
+
+import java.time.LocalDate;
+import java.time.Month;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class Product {
+    ProductType category;
+    double price;
+    boolean canApplyDiscount;
+    LocalDate date;
+
+    public Product(ProductType category, float price, boolean canApplyDiscount, LocalDate date) {
+        this.category = category;
+        this.price = price;
+        this.canApplyDiscount = canApplyDiscount;
+        this.date = date;
+    }
+
+    public ProductType getCategory() {
+        return category;
+    }
+
+    public double getPrice() {
+        return price;
+    }
+
+    public boolean isCanApplyDiscount() {
+        return canApplyDiscount;
+    }
+
+    public LocalDate getDate() {
+        return date;
+    }
+
+    public void setCategory(ProductType category) {
+        this.category = category;
+    }
+
+    public void setPrice(double price) {
+        this.price = price;
+    }
+
+    public void setCanApplyDiscount(boolean canApplyDiscount) {
+        this.canApplyDiscount = canApplyDiscount;
+    }
+
+    public void setDate(LocalDate date) {
+        this.date = date;
+    }
+
+    @Override
+    public String toString() {
+        return "Product{" +
+                "category=" + category +
+                ", price=" + price +
+                ", canApplyDiscount=" + canApplyDiscount +
+                ", date=" + date +
+                '}';
+    }
+
+    public static List<Product> getHighPriceBook(List<Product> productList) {
+        var stream = productList.stream()
```

Comment:
```suggestion
        return productList.stream()
            ....
            .toList();
```
і так і так можна, але це чистіше буде